webapp-django/crashstats/crashstats/static/crashstats/js/socorro/oauth2.js

```diff
@@ -110,8 +121,18 @@ var OAuth2 = (function() {
                         $.post(url, data)
                         .done(function(response) {
                             // It worked!
-                            // TODO: https://bugzilla.mozilla.org/show_bug.cgi?id=1283296
-                            document.location.reload();
+                            var next = getParameterByName('next');
+                            // only if ?next=/... exists on the current URL
+                            if (next) {
+                                // A specific URL exits.
+                                // This is most likely the case when you tried
+                                // to access a privileged URL whilst being
+                                // anonymous and being redirected.
+                                // Make sure it's on this server
+                                document.location.pathname = next;
+                            } else {
```

Comment:
We have the ``Qs`` library in Socorro, so you can use it like this: 

```javascript
Qs.parse(document.location.search.slice(1));
// Object { product: "Firefox", _dont_run: "1" }
```

This is compatible with most browsers, and is already in our base bundle, so it's accessible anywhere in the webapp. 

Documentation: https://github.com/ljharb/qs/